debugger/IController.hx

```diff
@@ -162,27 +162,24 @@ enum Command
     // ErrorEvaluatingExpression
 }
 
-
 /**
- * A list of strings
+ * An array of strings
  **/
-enum StringList
-{
-    Terminator;
-    Element(string : String, next : StringList);
-}
+typedef StringArray = Array<String>;
 
 
 /**
- * A list of class, possibly truncated so as not to be too large.
- * If truncated, a subsequent query for the remainder of the list can be done
- * using the continued expression.
+ * An array of classes. It's not being truncated. If this becomes an issue, it should
+ * be solved by adding a parameter that could be used to get the remainder of the array.
+ * It would have been much better if we had structs available in place of classes.
```

Comment:
Structs in place of classes would have been no better.  You still need to handle arrays that are too large and enums would make this no more difficult than full classes would have.  Furthermore, haxe has built in support for serializing enums, I do not believe that it has such support for serializing classes.  That's the whole reason that enums were used instead of classes.